stackslib/src/chainstate/nakamoto/tests/mod.rs

```diff
@@ -1636,3 +1643,272 @@ pub fn test_get_highest_nakamoto_tenure() {
+                    winning_block_commit,
+                )]
+            } else {
+                vec![]
+            }
+        };
+
+        {
+            let mut tx = SortitionHandleTx::begin(sort_db, &last_snapshot.sortition_id).unwrap();
+            let _index_root = tx
+                .append_chain_tip_snapshot(
+                    &last_snapshot,
+                    &snapshot,
+                    &winning_ops,
+                    &vec![],
+                    None,
+                    None,
+                    None,
+                )
+                .unwrap();
+            tx.test_update_canonical_stacks_tip(
+                &snapshot.sortition_id,
+                &snapshot.canonical_stacks_tip_consensus_hash,
+                &snapshot.canonical_stacks_tip_hash,
+                snapshot.canonical_stacks_tip_height,
+            )
+            .unwrap();
+            tx.commit().unwrap();
+        }
+
+        last_snapshot = SortitionDB::get_block_snapshot(sort_db.conn(), &snapshot.sortition_id)
+            .unwrap()
+            .unwrap();
+
+        if i > 3 {
+            // have block-commit
+            // check the stackerdb config as of this chain tip
+            let stackerdb_config =
+                NakamotoChainState::make_miners_stackerdb_config(sort_db).unwrap();
+            eprintln!(
+                "stackerdb_config at i = {} (sorition? {}): {:?}",
+                &i, sortition, &stackerdb_config
+            );
+
+            stackerdb_configs.push(stackerdb_config);
+
+            // make a stackerdb chunk for a hypothetical block
+            let header = NakamotoBlockHeader {
+                version: 1,
+                chain_length: 2,
+                burn_spent: 3,
+                consensus_hash: ConsensusHash([0x04; 20]),
+                parent_block_id: StacksBlockId([0x05; 32]),
+                tx_merkle_root: Sha512Trunc256Sum([0x06; 32]),
+                state_index_root: TrieHash([0x07; 32]),
+                miner_signature: MessageSignature::empty(),
+                signer_signature: ThresholdSignature::mock(),
+            };
+            let block = NakamotoBlock {
+                header,
+                txs: vec![],
+            };
+            if sortition {
+                let chunk = NakamotoBlockBuilder::make_stackerdb_block_proposal(
+                    &sort_db,
+                    i as u32,
+                    &block,
+                    &miner_keys[i - 3],
+                )
+                .unwrap()
+                .unwrap();
+                assert_eq!(chunk.slot_version, (i + 1) as u32);
+                assert_eq!(chunk.data, block.serialize_to_vec());
+                stackerdb_chunks.push(chunk);
+            } else {
+                assert!(NakamotoBlockBuilder::make_stackerdb_block_proposal(
+                    &sort_db,
+                    i as u32,
+                    &block,
+                    &miner_keys[i - 3]
+                )
```

Comment:
Should the prior two miners still be able to create block proposals?

Reply:
Why prior two?  I think we just want the prior